### ISSUE TITLE
Painting

### ISSUE THREAD

**Issue by FlorianRappl** [PorterX2Pro/Velocity#4]
For actually painting I suggest to have a closer look at CrossGraphics:  https://github.com/praeclarum/CrossGraphics - may be a good solution for going x-plat!

**Comment by PorterX2Pro** [comment]
This looks promising... I would second going for this library...

**Comment by PorterX2Pro** [comment]
Florian, so for rendering, Im thinking that we have one task that will calculate the drawing required, then generate an Action with the drawing code intact, THEN we pass that action over to the ui thread and execute it their, so no calculations are done on the ui thread.

**Comment by FlorianRappl** [comment]
Yes, this is the concept. Rendering should be a two stage process: calculate everything and actually draw it. Only the latter has to be done on the UI thread. The calculation has only to be changed / updated in case of viewport (e.g. window) or style (e.g. by manipulation via a script) changes.

**Comment by PorterX2Pro** [comment]
Exactly, so looking more into crossgraphics I'm gonna try to find a more advanced, and updated library for graphics.
